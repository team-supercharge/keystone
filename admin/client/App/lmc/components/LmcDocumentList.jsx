import React from 'react'
import PropTypes from 'prop-types'
import LmcDocumentItem from './LmcDocumentItem.jsx'

const LmcDocumentList = ({ documents, onDelete, listId }) => {
    return (
        <div>
            { Object.keys(documents).map(categoryName => (
                <div key={categoryName}>
                    <h2 style={styles.categoryName}>
                        {categoryName}
                    </h2>
                    <div className='lmc-theme-gradient' style={styles.divider} />
                    <ul style={styles.list}>
                        { documents[categoryName].map((document, i) => (
                            <LmcDocumentItem
                                key={i}
                                data={document}
                                listId={listId}
                                onDelete={onDelete}
                            />
                        )) }
                    </ul>
                </div>
            )) }
        </div>
    )
}

const styles = {
    categoryName: {
        marginBottom: '0.3em',
        fontWeight: 300,
    },
    divider: {
        height: 2,
        marginBottom: 22,
        width: '100%',
    },
    list: {
        listStyle: 'none',
        listStyleType: 'none',
        padding: 0,
    }
}

LmcDocumentList.propTypes = {
    documents: PropTypes.object,
    listId: PropTypes.string.isRequired,
    onDelete: PropTypes.func.isRequired
}

export default LmcDocumentList