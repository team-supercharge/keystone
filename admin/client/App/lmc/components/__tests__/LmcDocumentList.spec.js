import React from 'react'
import { shallow } from 'enzyme'
import LmcDocumentList from '../LmcDocumentList.jsx'

describe('LmcDocumentList', () => {
    let wrapper
    let documents
    const onDeleteMock = jest.fn()

    beforeEach(() => {
        documents = { 
            'CategoryName1': [{ name: 'TestDocument' }],
            'CategoryName2': [{ name: 'TestDocument2' }]
        }

        wrapper = shallow(
            <LmcDocumentList
                documents={documents}
                listId='testId'
                onDelete={onDeleteMock}
            />
        )
    })

    it('renders correctlty', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('renders the correct number of LmcDocumentItems', () => {
        expect(wrapper.find('LmcDocumentItem').length).toEqual(2)
    })

    it('renders the category names as headings', () => {
        const heading1 = wrapper.find('h2').first()
        const heading2 = wrapper.find('h2').at(1)

        expect(heading1.text()).toEqual('CategoryName1')
        expect(heading2.text()).toEqual('CategoryName2')
    })
})