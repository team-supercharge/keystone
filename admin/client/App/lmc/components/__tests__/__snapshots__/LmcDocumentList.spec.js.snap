// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcDocumentList renders correctlty 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
      style={
        Object {
          "marginBottom": 40,
        }
      }
    >
      <h2
        style={
          Object {
            "fontWeight": 300,
            "hyphens": "auto",
            "marginBottom": "0.3em",
            "textOverflow": "ellipsis",
          }
        }
      >
        CategoryName1
      </h2>
      <div
        className="lmc-theme-gradient"
        style={
          Object {
            "height": 2,
            "marginBottom": 22,
            "width": "100%",
          }
        }
      />
      <ul
        style={
          Object {
            "listStyle": "none",
            "listStyleType": "none",
            "padding": 0,
          }
        }
      >
        <div>
          <LmcDocumentItem
            data={
              Object {
                "name": "TestDocument",
              }
            }
            listId="testId"
            onDelete={[MockFunction]}
          />
          <div
            style={
              Object {
                "backgroundColor": "#f2f2f2",
                "height": 2,
                "marginBottom": 12,
                "marginTop": 8,
              }
            }
          />
        </div>
      </ul>
    </div>
    <div
      style={
        Object {
          "marginBottom": 40,
        }
      }
    >
      <h2
        style={
          Object {
            "fontWeight": 300,
            "hyphens": "auto",
            "marginBottom": "0.3em",
            "textOverflow": "ellipsis",
          }
        }
      >
        CategoryName2
      </h2>
      <div
        className="lmc-theme-gradient"
        style={
          Object {
            "height": 2,
            "marginBottom": 22,
            "width": "100%",
          }
        }
      />
      <ul
        style={
          Object {
            "listStyle": "none",
            "listStyleType": "none",
            "padding": 0,
          }
        }
      >
        <div>
          <LmcDocumentItem
            data={
              Object {
                "name": "TestDocument2",
              }
            }
            listId="testId"
            onDelete={[MockFunction]}
          />
          <div
            style={
              Object {
                "backgroundColor": "#f2f2f2",
                "height": 2,
                "marginBottom": 12,
                "marginTop": 8,
              }
            }
          />
        </div>
      </ul>
    </div>
  </div>,
  "nodes": Array [
    <div>
      <div
        style={
          Object {
            "marginBottom": 40,
          }
        }
      >
        <h2
          style={
            Object {
              "fontWeight": 300,
              "hyphens": "auto",
              "marginBottom": "0.3em",
              "textOverflow": "ellipsis",
            }
          }
        >
          CategoryName1
        </h2>
        <div
          className="lmc-theme-gradient"
          style={
            Object {
              "height": 2,
              "marginBottom": 22,
              "width": "100%",
            }
          }
        />
        <ul
          style={
            Object {
              "listStyle": "none",
              "listStyleType": "none",
              "padding": 0,
            }
          }
        >
          <div>
            <LmcDocumentItem
              data={
                Object {
                  "name": "TestDocument",
                }
              }
              listId="testId"
              onDelete={[MockFunction]}
            />
            <div
              style={
                Object {
                  "backgroundColor": "#f2f2f2",
                  "height": 2,
                  "marginBottom": 12,
                  "marginTop": 8,
                }
              }
            />
          </div>
        </ul>
      </div>
      <div
        style={
          Object {
            "marginBottom": 40,
          }
        }
      >
        <h2
          style={
            Object {
              "fontWeight": 300,
              "hyphens": "auto",
              "marginBottom": "0.3em",
              "textOverflow": "ellipsis",
            }
          }
        >
          CategoryName2
        </h2>
        <div
          className="lmc-theme-gradient"
          style={
            Object {
              "height": 2,
              "marginBottom": 22,
              "width": "100%",
            }
          }
        />
        <ul
          style={
            Object {
              "listStyle": "none",
              "listStyleType": "none",
              "padding": 0,
            }
          }
        >
          <div>
            <LmcDocumentItem
              data={
                Object {
                  "name": "TestDocument2",
                }
              }
              listId="testId"
              onDelete={[MockFunction]}
            />
            <div
              style={
                Object {
                  "backgroundColor": "#f2f2f2",
                  "height": 2,
                  "marginBottom": 12,
                  "marginTop": 8,
                }
              }
            />
          </div>
        </ul>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <LmcDocumentList
        documents={
          Object {
            "CategoryName1": Array [
              Object {
                "name": "TestDocument",
              },
            ],
            "CategoryName2": Array [
              Object {
                "name": "TestDocument2",
              },
            ],
          }
        }
        listId="testId"
        onDelete={[MockFunction]}
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "documents": Object {
            "CategoryName1": Array [
              Object {
                "name": "TestDocument",
              },
            ],
            "CategoryName2": Array [
              Object {
                "name": "TestDocument2",
              },
            ],
          },
          "listId": "testId",
          "onDelete": [MockFunction],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
            style={
              Object {
                "marginBottom": 40,
              }
            }
          >
            <h2
              style={
                Object {
                  "fontWeight": 300,
                  "hyphens": "auto",
                  "marginBottom": "0.3em",
                  "textOverflow": "ellipsis",
                }
              }
            >
              CategoryName1
            </h2>
            <div
              className="lmc-theme-gradient"
              style={
                Object {
                  "height": 2,
                  "marginBottom": 22,
                  "width": "100%",
                }
              }
            />
            <ul
              style={
                Object {
                  "listStyle": "none",
                  "listStyleType": "none",
                  "padding": 0,
                }
              }
            >
              <div>
                <LmcDocumentItem
                  data={
                    Object {
                      "name": "TestDocument",
                    }
                  }
                  listId="testId"
                  onDelete={[MockFunction]}
                />
                <div
                  style={
                    Object {
                      "backgroundColor": "#f2f2f2",
                      "height": 2,
                      "marginBottom": 12,
                      "marginTop": 8,
                    }
                  }
                />
              </div>
            </ul>
          </div>
          <div
            style={
              Object {
                "marginBottom": 40,
              }
            }
          >
            <h2
              style={
                Object {
                  "fontWeight": 300,
                  "hyphens": "auto",
                  "marginBottom": "0.3em",
                  "textOverflow": "ellipsis",
                }
              }
            >
              CategoryName2
            </h2>
            <div
              className="lmc-theme-gradient"
              style={
                Object {
                  "height": 2,
                  "marginBottom": 22,
                  "width": "100%",
                }
              }
            />
            <ul
              style={
                Object {
                  "listStyle": "none",
                  "listStyleType": "none",
                  "padding": 0,
                }
              }
            >
              <div>
                <LmcDocumentItem
                  data={
                    Object {
                      "name": "TestDocument2",
                    }
                  }
                  listId="testId"
                  onDelete={[MockFunction]}
                />
                <div
                  style={
                    Object {
                      "backgroundColor": "#f2f2f2",
                      "height": 2,
                      "marginBottom": 12,
                      "marginTop": 8,
                    }
                  }
                />
              </div>
            </ul>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div
            style={
              Object {
                "marginBottom": 40,
              }
            }
          >
            <h2
              style={
                Object {
                  "fontWeight": 300,
                  "hyphens": "auto",
                  "marginBottom": "0.3em",
                  "textOverflow": "ellipsis",
                }
              }
            >
              CategoryName1
            </h2>
            <div
              className="lmc-theme-gradient"
              style={
                Object {
                  "height": 2,
                  "marginBottom": 22,
                  "width": "100%",
                }
              }
            />
            <ul
              style={
                Object {
                  "listStyle": "none",
                  "listStyleType": "none",
                  "padding": 0,
                }
              }
            >
              <div>
                <LmcDocumentItem
                  data={
                    Object {
                      "name": "TestDocument",
                    }
                  }
                  listId="testId"
                  onDelete={[MockFunction]}
                />
                <div
                  style={
                    Object {
                      "backgroundColor": "#f2f2f2",
                      "height": 2,
                      "marginBottom": 12,
                      "marginTop": 8,
                    }
                  }
                />
              </div>
            </ul>
          </div>
          <div
            style={
              Object {
                "marginBottom": 40,
              }
            }
          >
            <h2
              style={
                Object {
                  "fontWeight": 300,
                  "hyphens": "auto",
                  "marginBottom": "0.3em",
                  "textOverflow": "ellipsis",
                }
              }
            >
              CategoryName2
            </h2>
            <div
              className="lmc-theme-gradient"
              style={
                Object {
                  "height": 2,
                  "marginBottom": 22,
                  "width": "100%",
                }
              }
            />
            <ul
              style={
                Object {
                  "listStyle": "none",
                  "listStyleType": "none",
                  "padding": 0,
                }
              }
            >
              <div>
                <LmcDocumentItem
                  data={
                    Object {
                      "name": "TestDocument2",
                    }
                  }
                  listId="testId"
                  onDelete={[MockFunction]}
                />
                <div
                  style={
                    Object {
                      "backgroundColor": "#f2f2f2",
                      "height": 2,
                      "marginBottom": 12,
                      "marginTop": 8,
                    }
                  }
                />
              </div>
            </ul>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcDocumentList
    documents={
      Object {
        "CategoryName1": Array [
          Object {
            "name": "TestDocument",
          },
        ],
        "CategoryName2": Array [
          Object {
            "name": "TestDocument2",
          },
        ],
      }
    }
    listId="testId"
    onDelete={[MockFunction]}
  />,
}
`;
